tests/v2/validation/workloads/workload_test.go

```diff
@@ -192,6 +193,91 @@ func (w *WorkloadTestSuite) TestWorkloadStatefulset() {
 	require.NoError(w.T(), err)
 }
 
+func (w *WorkloadTestSuite) TestWorkloadUpgrade() {
+	subSession := w.session.NewSession()
+	defer subSession.Cleanup()
+
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(w.client, w.cluster.ID)
+	require.NoError(w.T(), err)
+
+	upgradeDeployment, err := deployment.CreateDeployment(w.client, w.cluster.ID, namespace.Name, 2, "", "", false, false)
+	require.NoError(w.T(), err)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"1"}, "nginx", 2)
+
+	containerName := namegen.AppendRandomString("updatetestcontainer")
+	newContainerTemplate := workloads.NewContainer(containerName,
+		"redis",
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+	upgradeDeployment.Spec.Template.Spec.Containers = []corev1.Container{newContainerTemplate}
+
+	upgradeDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment)
+	require.NoError(w.T(), err)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"1", "2"}, "redis", 2)
+
+	containerName = namegen.AppendRandomString("updatetestcontainertwo")
+	newContainerTemplate = workloads.NewContainer(containerName,
+		"ubuntu",
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+	newContainerTemplate.TTY = true
+	newContainerTemplate.Stdin = true
+	upgradeDeployment.Spec.Template.Spec.Containers = []corev1.Container{newContainerTemplate}
+
+	_, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment)
+	require.NoError(w.T(), err)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"1", "2", "3"}, "ubuntu", 2)
+
+	logRollback, err := deployment.RollbackDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment, 1)
+	require.NoError(w.T(), err)
+	require.NotEmpty(w.T(), logRollback)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"2", "3", "4"}, "nginx", 2)
+
+	logRollback, err = deployment.RollbackDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment, 2)
+	require.NoError(w.T(), err)
+	require.NotEmpty(w.T(), logRollback)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"3", "4", "5"}, "redis", 2)
+
+	logRollback, err = deployment.RollbackDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment, 3)
+	require.NoError(w.T(), err)
+	require.NotEmpty(w.T(), logRollback)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"4", "5", "6"}, "ubuntu", 2)
```

Comment:
once we change from command to v1 or public api to get the deployment and then revision history, we wouldn't need to send in the values as a list of strings here.

Reply:
Yes, I can do this change, but it would not be the same test of test_wl_upgrade